Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -1403,6 +1403,21 @@ def visit_DefNode(self, node):
         if decorator_node is not None:
             decorator = decorator_node.decorator
             if decorator.is_name:
+                # If there is a DefNode with the same name, that is probably a mistake like
+                # cdef class A:
+                #    def p(self):
+                #       return self.value
+                #
+                #    @property
+                #    def p(self):
+                #        return self.p
+                #
+                # In a python class this will raise a runtime RecursionError
+                same_name = [n for n in self.scope_node.body.stats 
+                             if getattr(n,'name', '') == node.name and n is not node]
+                if len(same_name) > 0:
```

Comment:
Any reason you're not using `any()`?
```suggestion
                if any(n is not node and getattr(n,'name', '') == node.name
                       for n in self.scope_node.body.stats):
```

Reply:
adopted